docs/proposal-0002-moving-metrics-out-of-mozanalysis.md

```diff
@@ -0,0 +1,63 @@
+# Moving metrics out of mozanalysis into jetstream-config
+
+* Status: Accepting feedback
+* Date: 2022-07-18
+
+Related issue: https://github.com/mozilla/jetstream/issues/1131
+
+## Context and Problem Statement
+
+Metric definitions that can be referenced and reused across experiment configurations are currently part of mozanalysis. 
+To avoid having conflicting or duplicate metric definitions, we would like to move all definitions into jetstream-config. This would also make adding new metrics significantly easier that should be computed for multiple experiments as it would no longer be necessary to build and publish a new version of mozanalysis.
+
+## Decision Drivers
+
+* The changes should not break existing code that uses mozanalysis
+* Metrics should be defined in a single place
+* It should still be possible to reference metrics in configuration files without having to redefine their definitions
+
+## Implementation
+
+### jetstream-config Changes
+
+Metric definitions will be stored in the [jetstream-config](https://github.com/mozilla/jetstream-config) repository. A new `definitions/` directory will be added. Metric definitions will be collected for each platform in separate files:
+
+```
+defaults/
+  └ ...
+definitions/
+  └ firefox_desktop.toml
+  └ fenix.toml
+  └ ...
+outcomes/
+  └ ...
+1-click-pin-experiment.toml
+...
+```
+
+The format of these files follows the default configuration file format, however no experiment specific configuation details need to be specified. All metrics that are defined in configuration files located under `definitions/` can be referenced by their metric slug in other configuration files.
+
+The `definitions/` directory will be added to the `CODEOWNERS` file which ensures that any changes will need to be reviewed and approved before merging. After changes have been merged to the definitions, experiments **do not** automatically get rerun (unless their configuration file has also been updated).
+
+
+### jetstream Changes
+
+The code for parsing jetstream configuration files will be moved into a separate Python library. This library will be used by mozanalysis, jetstream, the jetstream docs generator (and maybe opmon? in the future) for pulling in and parsing existing config files.
```

Comment:
I think it could be more confusing for users that just want to add some config files but accidentally get somehow lost inside the code. Plus, we might, for example, have multiple config repos, like a jetstream-config-private repo.

Reply:
Ohh yeah, private config repos… that’s a great point. Agree, let’s keep it separate then